### PR TITLE
hey watch --notify: the engine under the Omarchy bar plugin, in place of hey omarchy

### PR DESCRIPTION
**Pull request by jeremy**
Supersedes #236.

## What

`basecamp/omarchy-hey-plugin` (Jason, Rob) shells out to `hey` for the Imbox; #236 gave it `hey omarchy poll`, a ten-minute poll that also diffed a fingerprint file to toast new mail. Stanko asked why not `hey watch`; the sharper objection is that an Omarchy-named, watch-like subcommand sullied the CLI surface — and a poll is ten minutes stale by construction (the TUI→IPC nudge in #236 covered only the TUI's own mutations). So, with generic commands only:

- **`hey box imbox --json` is the read, `hey watch` is the wake-up.** The plugin runs one `hey watch` per shell (it becomes a `service`-kind plugin), re-reads the Imbox per batch of events, and `hey watch` already catches up from its cursor on reconnect and skips ahead on a 409 — live and self-healing, no IPC nudge: the cable reports the TUI's own mutations back within a second. The watch covers every box because a move out of the Imbox is written in the target box's feed, never the Imbox's.
- **`hey watch` says what state it is in.** Three new lines beside the thread changes: `ready` once every box is caught up and the subscription is live (again after each reconnect's catch-up; a read that failed is retried and `ready` waits for it) — a reader that reads its own state on that line has no startup gap to reason about; `disconnected` when the cable drops; `resync` when a box changed more than the feed can list and the watch skipped ahead (this used to reach stderr only). A resync is a change (scripts run, `--exit-on-first` counts it); ready and disconnected are stdout-only and carry no `box`.
- **New mail is a watch event.** Every `added`/`updated` line carries `"new": true|false`, and `--events new` selects the true ones — a union with `added`, `updated`, `deleted` (`new` alone = new mail only; `added,new` = arrivals plus new activity on known threads). `--run-*` scripts see `HEY_NEW=1`; `hey watch --box imbox --events new --exit-on-first` is "block until new mail". New = an unseen, unmuted posting whose `active_at` is later than the watch last recorded for it, or later than the watch's start — on HEY's clock (`serverNow`) — for a thread without a record; `active_at` moves on new mail only, so seen flips, mutes and moves are never new, a box's first read carries its backlog quietly, mail that arrived during startup is new, and a box that skipped ahead on a 409 gets its own cutoff at the cursor it skipped to. The tracker records every posting the watch reads, in every box and whatever `--events` reports. **No state file.** (An earlier cut of this PR sent the toast itself as `hey watch --notify`; Stanko's review — why not `--run-async`? — showed that flag bundled HEY semantics with a desktop's presentation. The semantics stayed here; the toast moved to the plugin, and nothing desktop-shaped is left in `watch*.go`.)
- **The `hey omarchy` group is gone** (never released: v0.1.1 predates all of it and `bar-status` was hidden): `bar-status`, #236's `poll`, the poll state and flock, the config-degraded signalling only the poll reported, the account-scope and local-config exemptions. `hey setup omarchy` keeps #236's plugin-entry configuration and legacy `hey-unread` removal; turning toasts on is now just setting the key.

Main refreshed the cable credentials per dial itself meanwhile (`WithHeaderFunc`, 45b0263); this branch is rebased on it and the transport wrapper it carried is gone.

Docs: `docs/omarchy.md` ("The bar plugin", "New mail is a watch event", "The plugin toasts", a **Generic commands; the face presents** decision, follow-ups), README, the skill.

Companion: basecamp/omarchy-hey-plugin#3 (service kind, watch-first with the read on `ready`, debounced re-read, backoff restart, auth exit waits for the probe, and the toast: Imbox `new` lines debounced into one `omarchy-notification-send --app-name HEY … -p [-r <id>]`).

## Release note

Unchanged from #236: do **not** release 0.2.0 until the plugin repo is public and its PR is merge-approved.

Heads-up, not this PR's: `make check-surface-compat` fails on `main` today because `.surface-breaking` lists `hey screener list --count`, which v0.1.1 never had (the script calls that entry stale). Dropping that line fixes it.

## Verified

`make build && make test && make lint`; `.surface` removes nothing released (`--notify`/`--notify-box` existed only on this branch). Live: `hey watch --box imbox --events new` against production connects, reads HEY's clock, prints `ready`, and exits clean on `--timeout`; an earlier cut reported a `hey seen` from another process within ~1 s, and the reconnect/token-refresh matrix passed on it. Plugin: `./tests/run` (23 node + 41 QML) and `omarchy plugin validate` green.

Not yet done live on the final cut: a plugin toast for mail from an external mailbox, a second one replacing it, `hey seen` → `new:false`, and SUPER+CTRL+comma silencing it.

### CODE REVIEW

File: internal/cmd/watch.go [hunk 1]
```diff
@@ -425,15 +441,26 @@ func (w *postingsWatch) readBox(ctx context.Context, box *watchedBox) error {
...
 	}
 
+	// The notification covers what was reported — the same --events filter, the
+	// same stop at the first change — and one read is one batch, so a burst of
+	// mail is one toast.
+	var added, updated []generated.Posting
 	for _, posting := range changes.Added {
-		w.report(ctx, watchEvent{Change: "added", At: watchTime(posting.CreatedAt), PostingID: posting.Id}, box, &posting)
+		if w.report(ctx, watchEvent{Change: "added", At: watchTime(posting.CreatedAt), PostingID: posting.Id}, box, &posting) {
+			added = append(added, posting)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in d5b291e. `readBox` now hands the notifier every added/updated posting it read (`observed`) alongside the ones it reported; `record(observed)` keeps each thread's `active_at` whatever `--events` says, and only the reported ones are candidates for a toast. `TestWatchNotifiesOnlyWatchedEvents` drives it end to end: `--events updated`, an arrival (unreported, recorded), then its seen flip (reported, not new).

File: internal/cmd/watch.go [hunk 2]
```diff
@@ -425,15 +441,26 @@ func (w *postingsWatch) readBox(ctx context.Context, box *watchedBox) error {
...
-		w.report(ctx, watchEvent{Change: "updated", At: watchTime(posting.UpdatedAt), PostingID: posting.Id}, box, &posting)
+		if w.report(ctx, watchEvent{Change: "updated", At: watchTime(posting.UpdatedAt), PostingID: posting.Id}, box, &posting) {
+			updated = append(updated, posting)
+		}
 	}
 	for _, posting := range changes.Deleted {
 		w.report(ctx, watchEvent{Change: "deleted", At: watchTime(posting.DeletedAt), PostingID: posting.Id}, box, nil)
 	}
+	if w.notifier != nil {
+		w.notifier.batch(box, added, updated)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in d5b291e, by removing the seed rather than patching it. There is no first-batch flag any more: a thread without a record is measured against the watch's start time, so a first read that ended in a 409 leaves nothing to discard later, and the backlog stays quiet for the reason it should — it was active before the watch began. The 409 also now produces a `resync` event on stdout (it only reached stderr before), so a reader knows to re-read the box; `TestWatchReportsAResyncAfterSkippingAhead` covers it.

File: README.md [hunk 3]
```diff
@@ -432,23 +433,31 @@ read from `~/.local/state/omarchy/current/theme/`, and restyles live when you ru
...
 
-`--notify` turns on new-mail toasts, off by default: the bar indicator's poll also sends
-at most one notification per interval — `Sender — Subject` for one new thread, `N new in
-Imbox` for more — replacing the previous toast rather than stacking, and clicking it
-focuses the TUI. Omarchy's notification silencing (SUPER+CTRL+comma) mutes them like any
-other app. See [docs/omarchy.md](docs/omarchy.md) for the details and what is planned next.
+`--notify` turns on new-mail toasts, off by default, by flipping the plugin's `notify`
+setting (the panel's toggle and `omarchy bar set 37signals.hey notify true --json` do the
+same); the plugin then runs `hey watch --notify`, which sends at most one notification per
+batch of changes — `Sender — Subject` for one new thread, `N new in Imbox` for more —
```
[the remark this answers is not in the document]
Comment (jeremy) [reply README.md]: Resolved the other way round in d5b291e: `--notify` now toasts the Imbox only, with `--notify-box` to name another watched box. The plugin watches every box (it has to, to see a thread leave the Imbox) but HEY's attention model puts new mail in one place, so its toasts are `N new in Imbox` exactly as the README says. `docs/omarchy.md` describes the general form, `N new in <box>`, and the default.

File: docs/omarchy.md [hunk 4]
```diff
@@ -149,8 +165,23 @@ fetch serves both the indicator and the toasts.
...
-- **No HTML scraping to feed widgets.** The indicator uses the same typed SDK read as
-  `hey box imbox`.
+- **No HTML scraping to feed widgets.** The plugin reads with `hey box imbox`, the same
+  typed SDK read as everyone else.
+- **CLI is the engine, plugin is the face.** The plugin owns rendering and settings; the
+  CLI owns what needs HEY's semantics — the changes cursor and catching up after a
+  disconnect, what counts as new, a toast that honors DND — where it is Go-tested once
+  instead of re-derived in QML. The plugin's panel does show an unread *number* per
+  account; the bar icon itself stays a glyph that lights or does not, and the toast never
+  carries the unread total either — `N new in Imbox` is how many threads arrived in one
```
[the remark this answers is not in the document]
Comment (jeremy) [reply docs/omarchy.md:211]: Same resolution as the README thread: `--notify` toasts the Imbox by default (d5b291e), so for the plugin the headline is `N new in Imbox`; the `hey watch --notify` section states the general `N new in <box>` and the `--notify-box` override. Added a "The Imbox, unless told otherwise" bullet there with the reasoning.

File: internal/cmd/watch_notify.go [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+func (n *desktopNotifier) batch(box *watchedBox, added, updated []generated.Posting) {
+	fresh := n.fresh(added, updated)
+
+	// A box's first read is its catch-up: the server's cursor is the box's last
+	// activity, not this moment, and --since moves it further back. What it
+	// carries is the backlog — recorded, so later updates measure against it,
+	// but never toasted.
+	if !n.seeded[box.id] {
+		n.seeded[box.id] = true
+		return
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch_notify.go]: Agreed and done in d5b291e: the unconditional first-batch return is gone. Every candidate is measured against the watch's recorded `active_at` for the thread, or against `started` when there is none, and `started` is taken before the cursors are fetched. `TestWatchNotifyKeepsTheBacklogQuietButNotMailSinceItBegan` has a first read carrying one backlog thread and one that arrived during startup: one toast, for the arrival.

File: internal/cmd/root.go [hunk 6]
```diff
@@ -207,7 +205,6 @@ func newRootCmd() *cobra.Command {
 	root.AddCommand(newIgnoreCommand().cmd)
 	root.AddCommand(newStopIgnoringCommand().cmd)
 	root.AddCommand(newSetupCommand().cmd)
-	root.AddCommand(newOmarchyCommand().cmd)
 	root.AddCommand(newTuiCommand().cmd)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/root.go:215]: Declining: there is no installed base to migrate. The last release is v0.1.1 (`c0d3441`, tagged 2026-08-20 11:31 -0700); every Omarchy commit (#220, #225, #222) merged at 20:25 the same day, `git show v0.1.1:.surface | rg omarchy` is empty, and `hey omarchy bar-status` was `Hidden` and never in a release — `make check-surface-compat` against v0.1.1 passes with no `.surface-breaking` entry for it. A shell.json that names `hey omarchy bar-status` can only have come from a dev build, and `hey setup omarchy` removes a legacy `hey-unread` module on sight regardless, carrying its notify choice over to the plugin entry.

File: internal/cmd/watch.go [hunk 7]
```diff
@@ -425,15 +441,26 @@ func (w *postingsWatch) readBox(ctx context.Context, box *watchedBox) error {
...
 
+	// The notification covers what was reported — the same --events filter, the
+	// same stop at the first change — and one read is one batch, so a burst of
+	// mail is one toast.
+	var added, updated []generated.Posting
 	for _, posting := range changes.Added {
-		w.report(ctx, watchEvent{Change: "added", At: watchTime(posting.CreatedAt), PostingID: posting.Id}, box, &posting)
+		if w.report(ctx, watchEvent{Change: "added", At: watchTime(posting.CreatedAt), PostingID: posting.Id}, box, &posting) {
+			added = append(added, posting)
+		}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in d5b291e together with the Copilot thread on the same lines: notifier state comes from every added/updated posting the read carried (`record(observed)`), `--events` only decides which of them may toast. `TestWatchNotifyRecordsWhatItDoesNotReport` and `TestWatchNotifiesOnlyWatchedEvents` cover the exact sequence you describe (filtered arrival, then a reported seen flip).

**Review by monorkin — COMMENTED** [review]
Great fixes!
Can we replace the --notify with a --run-async? It looks like it just shells out and the run commands are exactly for that.
Then we could make the box filter generic.

File: internal/cmd/watch.go [hunk 8]
```diff
@@ -325,6 +406,9 @@ func (w *postingsWatch) listen(ctx context.Context, subscription *actioncable.Su
 			if err := w.readEveryBox(ctx); err != nil {
 				return err
 			}
+			w.announce(watchReady)
+		case <-w.dropped:
+			w.announce(watchDisconnected)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in 30dfcab — drop and reconnect now travel through one ordered queue (`noteConnection` appends, one wake-up, `followConnection` drains in order), so a drop is always announced before the reconnect that followed it. `TestWatchAnnouncesADropBeforeTheReconnectThatFollowedIt` plays the exact interleaving.

File: internal/cmd/watch_notify.go [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+	}
+
+	_, omarchyErr := exec.LookPath("omarchy-launch-or-focus-tui")
+
+	return &desktopNotifier{
+		errOut:   errOut,
+		send:     runNotifySend,
+		omarchy:  omarchyErr == nil,
+		now:      time.Now,
+		started:  time.Now(),
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch_notify.go]: Fixed in 30dfcab — the cutoff now comes from HEY's clock: `serverNow` reads the `Date` header of one cheap, uncached request (a nonce query keeps it out of the SDK's GET cache) before the cursors are taken, rounded down to the second so the doubt falls on a toast for mail a moment old; the local clock is only the fallback when the server's can't be read. Agreed the cursor is not a substitute (it is the box's last activity). `TestServerNowReadsTheServersClock` / `...FallsBackToTheLocalClock`.

File: internal/cmd/omarchy.go [hunk 10]
```diff
@@ -374,119 +337,155 @@ func notifyDetail(notify bool) string {
...
 	}
 	bar, _ := shell["bar"].(map[string]any)
 	layout, _ := bar["layout"].(map[string]any)
-	if barLayoutModule(layout, omarchyBarModuleID) == nil {
-		return stepResult("bar indicator", path, false, nil, "", "absent")
+	legacy, _ := s.removeLegacyBarModule(shell, layout)
+	notified := false
+	if plugin := barLayoutModule(layout, omarchyBarPluginID); plugin != nil {
+		_, notified = plugin["notify"]
+		delete(plugin, "notify")
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/omarchy.go]: Fixed in 30dfcab — `--remove` no longer touches the plugin's `notify` key; the step reports `kept` and points at `--no-notify`, the explicit off switch. Setup cannot tell a preference it wrote from one the panel or `omarchy bar set` wrote, so it leaves it. Docs and the remove test updated.

**Comment by jeremy** [comment]
🤖 @monorkin — you were right, and zooming out made it clearer why. `--notify` bundled two things:

| Inside `--notify` | What it really is | Where it belongs |
|---|---|---|
| "new" = unseen, unmuted, `active_at` later than the watch last recorded for the thread, or than the watch's start on HEY's clock; remembered across every box so a move-in isn't new | **HEY semantics** — state across events and the server clock | the CLI, once, Go-tested |
| `notify-send -a HEY -u low -p -r <id>`, the Omarchy glyph/exec hints, replace-not-stack, `Sender — Subject` / `N new in Imbox`, one toast per read, Imbox-only via `--notify-box` | **presentation** — a desktop's business, Omarchy-shaped at that | the consumer: the plugin, or a user's `--run-async` |

My defence was about the first row; your question was about the second. So 881aa9e splits them rather than sugaring one over the other:

- **`new` is a watch event.** Every `added`/`updated` line carries `"new": true|false`; `--events new` selects the true ones and is a union with `added`, `updated`, `deleted` (`--events new` alone = new mail only; `added,new` = arrivals plus new activity on known threads). `--run-*` scripts see `HEY_NEW=1`. `hey watch --box imbox --events new --exit-on-first` is "block until new mail" — the agent primitive. The rule, the server clock and the record-every-box memory are unchanged and still tested (`watch_new.go` / `watch_new_test.go`).
- **The box filter is generic again:** `--box` is the only one. `--notify`, `--notify-box`, notify-send and every Omarchy hint are gone from the CLI (neither flag was released, so no `.surface-breaking` entry).
- **The plugin toasts from the lines it already reads** (basecamp/omarchy-hey-plugin#3): Imbox `new` lines, a short debounce so one read's burst is one toast, `omarchy-notification-send --app-name HEY -u low --exec … -p [-r <id>]` with the replace id kept ten minutes. Flipping `notify` no longer restarts the watch; it only gates the toast path.
- Any other desktop is the honest one-liner: `hey watch --box imbox --events new --run-async 'notify-send -a HEY "New mail in HEY"'`.

Docs (`docs/omarchy.md`, README, skill) and `hey setup omarchy --notify`'s help follow; the decision record now says "Generic commands; the face presents" and means it.

File: internal/cmd/watch.go [hunk 11]
```diff
@@ -111,21 +129,38 @@ func (c *watchCommand) run(cmd *cobra.Command, args []string) error {
...
+	var notifier *desktopNotifier
+	if c.notify {
+		notifier = newDesktopNotifier(cmd.ErrOrStderr(), serverNow(ctx))
+	}
+
 	boxes, err := c.watchedBoxes(ctx)
 	if err != nil {
 		return err
 	}
+	if c.notify {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Moot as of 881aa9e: `--notify` and `--notify-box` are gone. New mail is a watch event now — every added/updated line carries `"new": true|false`, `--events new` selects it, and `--box` is the only box filter — so there is no notification configuration left to accept without effect; the toast moved to the Omarchy plugin (basecamp/omarchy-hey-plugin#3).

File: internal/cmd/watch.go [hunk 12]
```diff
@@ -358,15 +440,49 @@ func (w *postingsWatch) closedError(ctx context.Context) error {
...
+func (w *postingsWatch) followConnection(ctx context.Context) error {
+	for _, connected := range w.drainTransitions() {
+		if !connected {
+			w.announce(watchDisconnected)
+			continue
+		}
+		if err := w.readEveryBox(ctx); err != nil {
+			return err
+		}
+		w.announce(watchReady)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in b703d18 — `ready` now waits for every box: `catchUp` announces it only once no read is left waiting (`readyOnceCaughtUp`), a retry that reads the last box owed announces it (`retryUnread`), and a drop in between cancels the debt since the reconnect's catch-up announces its own. `TestWatchReadyWaitsForAFailedCatchUpRead` and `TestWatchDropWhileCatchingUpCancelsTheReadyItOwed`.

File: internal/cmd/watch.go [hunk 13]
```diff
@@ -423,22 +539,40 @@ func (w *postingsWatch) readBox(ctx context.Context, box *watchedBox) error {
 
 	if changes.FullSyncRequired {
 		fmt.Fprintf(w.errOut, "notice: too much changed in %s to follow one change at a time — skipping ahead, read the box with `hey box %s`\n", box.name, box.kind)
-		return w.skipAhead(ctx, box)
+		if err := w.skipAhead(ctx, box); err != nil {
+			return err
+		}
+		w.report(ctx, watchEvent{Change: watchResync, At: watchTime(time.Now())}, box, nil)
+		return nil
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in b703d18 — the notifier is gone (new mail is a watch event now), but the same hole was in the tracker. `skipAhead` now calls `newMail.skippedTo(box, cursor)`: that box's cutoff moves up to the cursor it skipped to, which is HEY's `last_posting_activity_at` for the box and so bounds every thread active in the gap. Per box, not global, so another box's mail during the gap is unaffected. Residual, documented: a gap thread that then *moves* to another box is measured there and may read as new once — the resync line is the reader's cue to re-read either way. `TestNewMailAfterASkipAheadIsSinceTheSkip`.

File: internal/cmd/watch.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Resolving: the flag and its configuration no longer exist (881aa9e).

File: internal/cmd/watch.go [hunk 14]
```diff
@@ -358,15 +409,86 @@ func (w *postingsWatch) closedError(ctx context.Context) error {
...
+// where it was, and ready waits for it: a reader told ready while a box is
+// still behind would take the gap for a clean picture. From ready on, anything
+// after the cursors is reported: a reader that wants a gap-free picture reads
+// its own state now, not before.
+func (w *postingsWatch) catchUp(ctx context.Context) error {
+	if err := w.readEveryBox(ctx); err != nil {
+		return err
+	}
+	w.catchingUp = true
+	w.readyOnceCaughtUp()
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in 4f1468d — `readyOnceCaughtUp` now checks the transition queue (`dropQueued`) and withholds ready while a drop is waiting to be drained; the drain announces disconnected and cancels the debt, and the reconnect's catch-up announces its own. `TestWatchReadyYieldsToADropQueuedDuringTheCatchUp` queues the drop before the catch-up and expects disconnected then ready.

File: internal/cmd/watch.go [hunk 15]
```diff
@@ -643,25 +816,49 @@ func (e watchEvent) environment() []string {
...
+	}
+	if e.PostingID != 0 {
+		environment = append(environment, "HEY_POSTING_ID="+strconv.FormatInt(e.PostingID, 10))
 	}
 	if e.ThreadID != 0 {
 		environment = append(environment, "HEY_THREAD_ID="+strconv.FormatInt(e.ThreadID, 10))
 	}
+	if e.isNew() {
+		environment = append(environment, "HEY_NEW=1")
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go:920]: Fixed in 4f1468d — `scriptCommand` now strips every per-event variable (`HEY_CHANGE`, `HEY_AT`, `HEY_BOX_*`, `HEY_POSTING_ID`, `HEY_THREAD_ID`, `HEY_NEW`) from the inherited environment before adding the event's own, so an update that is not new never sees a parent's `HEY_NEW=1` and a deletion never sees a thread id; everything else (`HEY_TOKEN` and the rest) is kept. `TestWatchScriptDoesNotInheritAnotherEventsVariables`.

File: internal/cmd/watch.go [hunk 16]
```diff
@@ -113,6 +138,10 @@ func (c *watchCommand) run(cmd *cobra.Command, args []string) error {
 		ctx = timed
 	}
 
+	// New mail is measured against the watch's start, so that is taken before
+	// the boxes' cursors are read: nothing can then change between the two.
+	newMail := trackNewMail(serverNow(ctx))
+
 	boxes, err := c.watchedBoxes(ctx)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in a6e3160 — each box's initial cursor now starts no later than the watch did (`noLaterThan`): when the box's own cursor (its last posting activity) is later than `serverNow`, it is moved back to the start, so mail that landed between the clock read and the box-list read is read by the catch-up and, being later than the start, is new. `--since` still wins. `TestWatchedBoxesStartNoLaterThanTheWatchDid`.

File: internal/cmd/watch.go [hunk 17]
```diff
@@ -360,15 +417,98 @@ func (w *postingsWatch) closedError(ctx context.Context) error {
...
 	}
 }
 
+// followConnection acts on the queued transitions in order: a drop is announced,
+// a reconnect catches every box up and then announces ready. Order is what keeps
+// a reader's picture right — a reconnect that completed while a slow catch-up
+// held the loop must not have its earlier drop announced after its ready.
+func (w *postingsWatch) followConnection(ctx context.Context) error {
+	for _, connected := range w.drainTransitions() {
+		if !connected {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go:473]: Fixed in a6e3160 — `followConnection` now takes transitions one at a time (`nextTransition`) instead of draining the queue, so a drop queued behind a reconnect is still in the shared queue while the reconnect catches up and `dropQueued` sees it. `TestWatchReadyYieldsToADropQueuedBehindTheReconnect`: reconnect then drop queued together → disconnected alone, then ready on the next reconnect.

File: internal/cmd/watch.go [hunk 18]
```diff
@@ -360,15 +417,98 @@ func (w *postingsWatch) closedError(ctx context.Context) error {
...
+// where it was, and ready waits for it: a reader told ready while a box is
+// still behind would take the gap for a clean picture. From ready on, anything
+// after the cursors is reported: a reader that wants a gap-free picture reads
+// its own state now, not before.
+func (w *postingsWatch) catchUp(ctx context.Context) error {
+	if err := w.readEveryBox(ctx); err != nil {
+		return err
+	}
+	w.catchingUp = true
+	w.readyOnceCaughtUp()
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in a6e3160 — `readyOnceCaughtUp` takes the context and also declines when `finished()` (the catch-up reported the change `--exit-on-first` was waiting for) or `ctx.Err() != nil` (the read was cut short). `TestWatchDoesNotSayReadyOnItsWayOut` covers both.

**Review by monorkin — APPROVED** [review]

**Comment by jeremy** [comment]
Sol's clock-window finding, fixed in eabf979: `serverNow` now translates the Date header back to the request's start by the monotonic elapsed time (`at.Add(-time.Since(started))`), so mail that lands while HEY is answering is later than the start, its box cursor is moved back by `noLaterThan`, and the catch-up reads it as new. A slow or SDK-retried request only moves the start earlier, and the fallback is the local clock at the start.

Tests: `TestServerNowIsTheClockWhenTheRequestBeganNotWhenItWasAnswered` (a delayed response puts the start at least the delay before Date) and `TestWatchReadsMailThatLandedWhileItReadTheClock` (end to end — slow clock request, a posting landing inside it, the box cursor rewound to the start, `new: true`, then `ready`).

Also noted: the 409 cross-box case is the documented residual, and the skip-ahead test's "moved" posting stays in its box — my recap overstated it. `make lint`, `go test -race ./internal/cmd ./internal/cable`, `make test` green.

File: internal/cmd/watch.go [hunk 19]
```diff
@@ -254,11 +292,29 @@ func watchCursor(changesURL, since string) (hey.PostingChangesCursor, error) {
...
+// own is later. The server bakes the box's last posting activity into its
+// cursor, so mail that landed after the watch read HEY's clock and before it
+// read the box list is already behind the cursor: the feed would start after
+// it, and nothing would ever report it. Starting from the watch's own start
+// reads it as part of the catch-up instead — and it is new, since it is later
+// than the start. A cursor that cannot be read is left as it is.
+func noLaterThan(cursor hey.PostingChangesCursor, started time.Time) hey.PostingChangesCursor {
+	at, err := time.Parse(watchCursorTimeLayout, cursor.Since)
+	if err == nil && at.After(started) {
+		cursor.Since = started.UTC().Format(watchCursorTimeLayout)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go:316]: Fixed in bacf0ba — `serverNow` hands the start out through `cutoffBefore`: truncated to the millisecond (the resolution of every cursor and `active_at`) and one millisecond earlier, so it is strictly before the instant it stands for on both the server-clock path and the local fallback. Mail in the start's own millisecond is then read (cursor rewound by `noLaterThan`) and `isNew`. `TestCutoffBeforeIsAWholeMillisecondStrictlyBefore` covers the truncation, the boundary case, the classification and the cursor; the delayed-response test now also asserts a whole-millisecond result.

[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go:316]: Follow-up b1e1dba: the local-clock fallback test now allows for the cutoff's deliberate millisecond step (it had asserted `not before the local clock`); `make test` is green on that head.

File: internal/cmd/watch.go [hunk 20]
```diff
@@ -425,26 +601,41 @@ func (w *postingsWatch) readBox(ctx context.Context, box *watchedBox) error {
...
 	}
 	for _, posting := range changes.Updated {
-		w.report(ctx, watchEvent{Change: "updated", At: watchTime(posting.UpdatedAt), PostingID: posting.Id}, box, &posting)
+		w.report(ctx, watchEvent{Change: "updated", At: watchTime(posting.UpdatedAt), PostingID: posting.Id, New: w.classify(box, posting)}, box, &posting)
 	}
 	for _, posting := range changes.Deleted {
 		w.report(ctx, watchEvent{Change: "deleted", At: watchTime(posting.DeletedAt), PostingID: posting.Id}, box, nil)
 	}
+	w.newMail.record(changes.Added)
+	w.newMail.record(changes.Updated)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in c2bb571 — `classify` now records each posting as soon as it is decided (`newMail.record` takes one posting), so a posting carried twice by one read is new once; haystack's added/updated sections are disjoint, but the tracker no longer leans on that. `TestNewMailCarriedTwiceByOneReadIsNewOnce` replaces the batch-order test.

File: internal/cmd/watch_new.go [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+// this watch last saw the thread — or since the watch began, for a thread it
+// has no record of, because anything active before that was already there: the
+// backlog a box's first read carries from the server's cursor, or a thread that
+// merely moved in. active_at moves on new mail only, not when a thread is read,
+// muted or moved, so none of those is new and a reply on a known thread is.
+func (n *newMail) isNew(boxID int64, posting generated.Posting) bool {
+	last, known := n.activeAt[posting.Id]
+	if !known {
+		last = n.started
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch_new.go:105]: Fixed in c2bb571 — every box is followed now and `--box` only picks the boxes whose changes are reported (`watchedBox.reported`; a `--box` naming nothing is still not found), so the record sees activity everywhere. `TestWatchFollowsEveryBoxAndReportsTheOnesAskedFor` is your scenario: an unseen reply in The Feed under `--box imbox` (nothing reported), then the move into the Imbox → `updated`, `new: false`. Flag help, README, skill, docs/omarchy.md and AGENTS.md say so.

File: internal/cmd/watch.go [hunk 22]
```diff
@@ -360,15 +443,109 @@ func (w *postingsWatch) closedError(ctx context.Context) error {
...
+
+// readyOnceCaughtUp pays the ready a catch-up owes, once no box is behind —
+// and not while a drop is waiting to be acted on: the connection went away
+// during the reads, so the stream would say ready with the subscription
+// already down. The drop cancels the debt when it is drained, and the
+// reconnect's catch-up announces its own. Nor when the watch is on its way
+// out: a catch-up that reported the change --exit-on-first was waiting for,
+// or one cut short by an interrupt, is not a watch that is live.
+func (w *postingsWatch) readyOnceCaughtUp(ctx context.Context) {
+	if !w.catchingUp || len(w.unread) > 0 || w.dropQueued() || w.finished() || ctx.Err() != nil {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in 8c68f04 — `readyOnceCaughtUp` now takes `transitionsMu` around the look at the queue *and* the announcement (`dropQueued` is gone), so it is one critical section with `noteConnection`'s append: a drop either lands before the look and withholds the ready, or after the ready is out, in which case the stream's order is the order things happened. The cable goroutine waits on at most one line of output. `go test -race` over the watch tests is clean; the existing queued-drop tests still pass.

File: internal/cmd/watch.go [hunk 23]
```diff
@@ -199,24 +232,31 @@ func (c *watchCommand) watchedBoxes(ctx context.Context) (map[int64]*watchedBox,
...
 		}
 		if cursor.Since == "" {
 			continue
 		}
+		if c.since == "" {
+			cursor = noLaterThan(cursor, started)
+		}
 
-		watched[box.Id] = &watchedBox{id: box.Id, kind: box.Kind, name: box.Name, cursor: cursor}
+		watched[box.Id] = &watchedBox{id: box.Id, kind: box.Kind, name: box.Name, cursor: cursor, reported: c.watching(box)}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go:253]: Fixed in 8c68f04 — `stopWatching` now returns not-found when no *reported* box remains, rather than when no box remains; the ones followed only for the record don't keep the watch alive. `TestWatchGivesUpWhenTheLastReportedBoxHasGone`: `--box imbox` with The Feed still listed, the Imbox gone after a 409 → the watch ends with the not-found error.

File: internal/cmd/watch.go [hunk 24]
```diff
@@ -425,18 +611,27 @@ func (w *postingsWatch) readBox(ctx context.Context, box *watchedBox) error {
 
 	if changes.FullSyncRequired {
 		fmt.Fprintf(w.errOut, "notice: too much changed in %s to follow one change at a time — skipping ahead, read the box with `hey box %s`\n", box.name, box.kind)
-		return w.skipAhead(ctx, box)
+		if err := w.skipAhead(ctx, box); err != nil {
+			return err
+		}
+		w.report(ctx, watchEvent{Change: watchResync, At: watchTime(time.Now())}, box, nil)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cmd/watch.go]: Fixed in 010efb3 — `skipAhead` now returns whether it actually moved the cursor (`(bool, error)`), and `readBox` reports `resync` only when it did; a box that vanished from the list is dropped without a resync, since there is nothing to re-read. `TestWatchGivesUpOnABoxThatHasGone` now watches with `resync` on and asserts nothing is written for the vanished box while another reported box remains.
